src/api/GoalsAPI/index.ts

```diff
@@ -124,11 +111,17 @@ export const updateGoal = async (id: string, changes: object) => {
 
 export const archiveGoal = async (goal: GoalItem) => {
   const updatedGoalStatus = { status: 1 };
+  const parentGoal = await getGoal(goal.parentGoalId);
+  
+  if (parentGoal && parentGoal.collaboration.status !== 'none') {
```

Comment:
Apologies, there are some modifications in the schema of goalItem as our story changed a bit. but approving for now

but Use **collaboration.status === "accept"** as that's the case where we will be following the exception